Dockerfile

```diff
@@ -8,9 +8,11 @@ RUN apt-get -y update \
   && rm -rf /var/lib/apt/lists/*
 
 # Install Python dependencies
-COPY requirements_dev.txt /app/
+COPY poetry.lock pyproject.toml /app/
 WORKDIR /app
-RUN pip install -r requirements_dev.txt
+RUN --mount=type=cache,mode=0755,target=/root/.cache/pip pip install poetry==1.5.0
+RUN poetry config virtualenvs.create false
+RUN --mount=type=cache,mode=0755,target=/root/.cache/pypoetry poetry install --no-root
```

Comment:
No need to install and configure `poetry` every time the dependencies change:

```diff
- COPY poetry.lock pyproject.toml /app/
  WORKDIR /app
  RUN --mount=type=cache,mode=0755,target=/root/.cache/pip pip install poetry==1.5.0
  RUN poetry config virtualenvs.create false
+ COPY poetry.lock pyproject.toml /app/
  RUN --mount=type=cache,mode=0755,target=/root/.cache/pypoetry poetry install --no-root
```

Same for `.devcontainer/Dockerfile`

Reply:
Good catch!